internal/agent/device/spec/spec.go

```diff
@@ -61,14 +67,15 @@ type Manager interface {
 	Rollback(ctx context.Context, opts ...RollbackOption) error
 	// GetDesired returns the desired rendered device from the management API.
 	GetDesired(ctx context.Context) (*v1alpha1.Device, bool, error)
-	// CheckPolicy validates the update policy is ready to process.
-	CheckPolicy(ctx context.Context, policyType policy.Type, version string) error
+
 	status.Exporter
 }
 
 type PriorityQueue interface {
 	// Add adds a new spec to the scheduler
 	Add(ctx context.Context, spec *v1alpha1.Device)
+	// AddWithDelay adds a new spec to the scheduler with a specific next available time
+	AddWithDelay(ctx context.Context, spec *v1alpha1.Device, nextAvailable time.Time)
```

Comment:
question: in the case that there is an error in the policy lets say its set to run in 1 year. how do we fix that?

Reply:
In the current case where the Queue's size is 1, this isn't an issue as the delayed entry gets evicted. In a case in which we would want to allow applying multiple specs in order... I'm not entirely sure. Either you respect the spec or you allow a future spec to control the pace.